From 4a3a574a38111d1e60fc95c63129138d5904f258 Mon Sep 17 00:00:00 2001
From: salieri <maliya355@outlook.com>
Date: Fri, 22 Mar 2024 16:50:15 +0800
Subject: [PATCH 10/11] [Debian] action: allow to run the test suite installed
 before

rgb.txt can not be located via FIG2DEV_RGBFILE environment variable.
This allows one to run the test suite without the package being
installed before.
---
 fig2dev/colors.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/fig2dev/colors.c b/fig2dev/colors.c
index 754150f..980028d 100644
--- a/fig2dev/colors.c
+++ b/fig2dev/colors.c
@@ -730,8 +730,13 @@ read_colordb(void)
 	FILE		*fp;
 #define MAX_LINE	100
 	char		s[MAX_LINE], s1[MAX_LINE];
+	char		*rgbfile;
 
-	fp = fopen(RGB_FILE, "r");
+	rgbfile = getenv("FIG2DEV_RGBFILE");
+	if (rgbfile == NULL) {
+		rgbfile = RGB_FILE;
+	}
+	fp = fopen(rgbfile, "r");
 	if (fp == NULL) {
 		Xcolors = defaultXcolors;
 		numXcolors = sizeof(defaultXcolors) / sizeof(struct color_db);
-- 
2.34.1

